src/pybind/mgr/cephadm/module.py

```diff
@@ -1302,13 +1302,30 @@ def add_host(self, spec: HostSpec) -> str:
         return self._add_host(spec)
 
     @handle_orch_error
-    def remove_host(self, host):
-        # type: (str) -> str
+    def remove_host(self, host: str, force: bool = False) -> str:
         """
         Remove a host from orchestrator management.
 
         :param host: host name
-        """
+        :param force: if true then allows to delete host with daemons running
+        """
+        if not force:
+            # Verify if it is possible to remove the host safely
+            daemons = [(d.hostname, d.daemon_type, str(d.container_id))
+                       for d in self.cache.get_daemons() if d.hostname == host]
+            if daemons:
+                self.log.warning(f"Blocked {host} removal. Daemons running: {daemons}")
+                daemons_table = ""
+                daemons.insert(0, ("-" * 40, "-" * 20, "-" * 15))
+                daemons.insert(0, ("hostname", "type", "container id"))
+                for d in daemons:
+                    daemons_table += "{:<40} {:<20} {:<15}\n".format(d[0], d[1], d[2])
+
+                return "Not allowed to remove %s from cluster. " \
+                    "The following daemons are running in the host:" \
+                    "\n%s" % (host, daemons_table)
```

Comment:
I feel like the hostname field is unnecessary for this table. We already say which host it is in the message above the table and we end up just printing out the same thing in that column for every row in the table.

![image](https://user-images.githubusercontent.com/47704447/113009518-05719480-9146-11eb-8705-b9405c57e674.png)

Maybe the service name for the daemon would be more useful infomration?

Reply:
Done. Thanks @adk3798 .. I haven't find another field in daemon description to replace hostname (... the service name would be also redundant with the type).

Now the look is:
```
[ceph: root@cephlab2-node-00 /]# ceph orch host rm cephlab2-node-01
Not allowed to remove cephlab2-node-01 from cluster. The following daemons are running in the host:
type                 container id   
-------------------- ---------------
mon                  a4d640c22e7c   
crash                2dd89ca71823   
node-exporter        4519f520a8c1   
osd                  6e3a5190db37  
```